src/auth/auth-wall.js

```diff
@@ -1,12 +1,21 @@
 import i18n from '@dhis2/d2-i18n'
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, { useState } from 'react'
 import classes from '../App.module.css'
-import { FirstLaunchDialog } from '../components'
+import { AlertStack, FirstLaunchDialog } from '../components'
+import { useCreateDataStore } from '../hooks'
 import { useIsAuthorized } from './use-is-authorized'
 
 const AuthWall = ({ children }) => {
     const { hasAppAccess, hasNamespace } = useIsAuthorized()
+    const { mutate, error } = useCreateDataStore()
+    const [hasDatastoreAccess, setDatastoreAccess] = useState(hasNamespace)
+
+    const handleSave = () => {
+        mutate().then((response) =>
+            setDatastoreAccess(response.status === 'OK')
```

Comment:
NIT: Check the [HTTP response status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses) instead. Something like this: 
`setDatastoreAccess(200 <= response.httpStatusCode && response.httpStatusCode < 300)`